python/lsst/pipe/base/connections.py

```diff
@@ -408,45 +553,129 @@ class attribute must match a function argument name in the ``run``
-        }
-        # Extract the configured value corresponding to each connection
-        # variable. I.e. for each connection identifier, populate a override
-        # for the connection.name attribute
-        self._nameOverrides = {
-            name: getattr(config.connections, name).format(**templateValues)
-            for name in self.allConnections.keys()
-        }
+    This may be replaced or modified in ``__init__`` to change the dimensions
+    of the task.  After ``__init__`` it will be a `frozenset` and may not be
+    replaced.
+    """
+
+    inputs: set[str]
+    """Set with the names of all `connectionTypes.Input` connection attributes.
+
+    This is updated automatically as class attributes are added, removed, or
+    replaced in ``__init__``.  Removing entries from this set will cause those
+    connections to be removed after ``__init__`` completes, but this is
+    supported only for backwards compatibility; new code should instead just
+    delete the collection attributed directly.  After ``__init__`` this will be
+    a `frozenset` and may not be replaced.
+    """
+
+    prerequisiteInputs: set[str]
+    """Set with the names of all `~connectionTypes.PrerequisiteInput`
+    connection attributes.
+
+    See `inputs` for additional information.
+    """
+
+    outputs: set[str]
+    """Set with the names of all `~connectionTypes.Output` connection
+    attributes.
 
-        # connections.name corresponds to a dataset type name, create a reverse
-        # mapping that goes from dataset type name to attribute identifier name
-        # (variable name) on the connection class
-        self._typeNameToVarName = {v: k for k, v in self._nameOverrides.items()}
+    See `inputs` for additional information.
+    """
+
+    initInputs: set[str]
+    """Set with the names of all `~connectionTypes.InitInput` connection
+    attributes.
+
+    See `inputs` for additional information.
+    """
+
+    initOutputs: set[str]
+    """Set with the names of all `~connectionTypes.InitOutput` connection
+    attributes.
+
+    See `inputs` for additional information.
+    """
+
+    allConnections: Mapping[str, BaseConnection]
+    """Mapping holding all connection attributes.
+
+    This is a read-only view that is automatically updated when connection
+    attributes are added, removed, or replaced in ``__init__``.  It is also
+    updated after ``__init__`` completes to reflect changes in `inputs`,
+    `prerequisiteInputs`, `outputs`, `initInputs`, and `initOutputs`.
+    """
+
+    _allConnections: dict[str, BaseConnection]
+
+    def __init__(self, *, config: PipelineTaskConfig | None = None):
+        pass
+
+    def __setattr__(self, name: str, value: Any) -> None:
+        if isinstance(value, BaseConnection):
+            previous = self._allConnections.get(name)
+            try:
+                getattr(self, value._connection_type_set).add(name)
```

Comment:
you could put `alasMap` in the class dictionary, possibly as `__aliasMap` and use it here as well. I'm not saying you have to do this, just hoping to spark some ideas around this topic.